### PR TITLE
PUBDEV-6903: Add support for SPNEGO to h2odriver

### PR DESCRIPTION
**Pull request by honzasterba**

### CODE REVIEW

File: scripts/jenkins/groovy/kerberosCommands.groovy [hunk 1]
```diff
@@ -7,15 +7,24 @@
...
         case H2O_HADOOP_STARTUP_MODE_STANDALONE:
             return getCommandStandalone(stageConfig)
         default:
             error("Startup mode ${stageConfig.customData.mode} for H2O with Hadoop is not supported")
     }
 }
 
-private GString getCommandHadoop(final stageConfig) {
+private GString getCommandHadoop(final stageConfig, final spnegoAuth) {
+    loginArgs = ""
```
Comment (michal-raska) [inline scripts/jenkins/groovy/kerberosCommands.groovy]: I think there's a missing `def`

File: scripts/jenkins/groovy/hadoopStage.groovy [hunk 2]
```diff
@@ -53,6 +54,7 @@ def call(final pipelineContext, final stageConfig) {
 private String getMakeTargetSuffix(final stageConfig) {
     switch (stageConfig.customData.mode) {
         case H2O_HADOOP_STARTUP_MODE_HADOOP:
+        case H2O_HADOOP_STARTUP_MODE_HADOOP_SPNEGO:
```
Comment (michal-raska) [inline scripts/jenkins/groovy/hadoopStage.groovy]: Could we remove the fallthrough please?

File: scripts/jenkins/groovy/kerberosCommands.groovy (hunk 1 above)
Comment (honzasterba) [reply scripts/jenkins/groovy/kerberosCommands.groovy]: thanks, also the if was broken :)

File: scripts/jenkins/groovy/hadoopStage.groovy (hunk 2 above)
Comment (honzasterba) [reply scripts/jenkins/groovy/hadoopStage.groovy]: fixed

**Comment by Tagar** [comment]
Nice.

If we enable SPNEGO authentication, would it be possible to whitelist only certain users, or perhaps use an LDAP query to get list of their Active Directory groups, and then only whitelist only certain groups for authorization. 

Thank you

**Comment by honzasterba** [comment]
Yes, I believe all that should be possible, you assign a service principal to the H2O cluster and then within your kerberos kdc you can define which users are allowed to access given service.

**Comment by Tagar** [comment]
Thank you @honzasterba

**Pull request merged**